src/main/java/com/example/FinalProject/controller/BookController.java

```diff
@@ -0,0 +1,67 @@
+package com.example.FinalProject.controller;
+
+import com.example.FinalProject.dto.dtoGetBook;
+import com.example.FinalProject.dto.dtoPostBook;
+import com.example.FinalProject.entity.BookEntity;
+import com.example.FinalProject.services.BookService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+
+@RestController
+@RequestMapping("/api/books")
+public class BookController {
+        @Autowired
+        private BookService bookService;
+
+        @GetMapping()
+        public List<dtoGetBook> getAllBooks() {
+            return bookService.getAllBooks();
+        }
+
+        @GetMapping("/{id}")
+        public ResponseEntity<BookEntity> getBookById(
+                @PathVariable Long id
+        ){
+            BookEntity book = bookService.getBookById(id);
+            if (book != null) {
+                return ResponseEntity.ok(book);
+            } else {
+                return ResponseEntity.notFound().build();
+            }
+        }
+
```

Comment:
Please move this checker to the service layer. If you want to check the request class, it's fine.

    ```
    @GetMapping("/{id}")
        public ResponseEntity<BookEntity> getBookById(
                @PathVariable Long id
        ){
            return bookService.getBookById(id);
         
        }

```